nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitJson.java

```diff
@@ -80,7 +82,7 @@
                 description = "The number of split FlowFiles generated from the parent FlowFile"),
         @WritesAttribute(attribute = "segment.original.filename ", description = "The filename of the parent FlowFile")
 })
-
+@SystemResourceConsideration(resource = SystemResource.MEMORY)
```

Comment:
In this particular context, we are buffering the entirety of the FlowFile's content (as a JsonNode object), in addition to all of the generated FlowFile objects. A two-stage approach may well be necessary for lots of splits, but even then if the JSON is extremely large you could potentially run out of heap space.